faiss/gpu/impl/RaftIVFFlat.cu

```diff
@@ -71,26 +59,41 @@ RaftIVFFlat::RaftIVFFlat(
                   useResidual,
                   scalarQ,
                   interleavedLayout,
+                  // skip ptr allocations in base class (handled by RAFT
+                  // internally)
                   indicesOptions,
                   space) {
     FAISS_THROW_IF_NOT_MSG(
             indicesOptions == INDICES_64_BIT,
             "only INDICES_64_BIT is supported for RAFT index");
-    reset();
 }
 
 RaftIVFFlat::~RaftIVFFlat() {}
 
-/// Find the approximate k nearest neighbors for `queries` against
-/// our database
+void RaftIVFFlat::reserveMemory(idx_t numVecs) {
```

Comment:
Note that one can actually pre-allocate lists in RAFT. This is already used internally here: https://github.com/rapidsai/raft/blob/0c75bb1985c1fe9177cf18e9c42d071d7115af7b/cpp/include/raft/neighbors/detail/ivf_flat_build.cuh#L256-L262.

When using RAFT, one can add all the training vectors in one batch, and in that case it is not needed to `reserveMemory` in advance, since the `build` method will do that. But if the user would like to add vectors in batches, then reserving the memory might make sense.

Reply:
That raft method is only for an individual list. For our case, we want to pre-allocate for numVecs vecs, without knowing the sizes of the lists before hand. FAISS gpu proper estimates the size of each list as `numVecs / numLists`. If this is something we'd like to do for RAFT lists as well, I can make this change. Leaving it open for discussion for now.